nuxeo-runtime/nuxeo-runtime-stream/src/main/java/org/nuxeo/runtime/stream/StreamProbe.java

```diff
@@ -21,38 +21,94 @@
 import static org.nuxeo.lib.stream.computation.log.ComputationRunner.NUXEO_METRICS_REGISTRY_NAME;
 
+import java.text.SimpleDateFormat;
+import java.time.Duration;
+import java.util.Date;
+
+import org.nuxeo.runtime.api.Framework;
 import org.nuxeo.runtime.management.api.Probe;
 import org.nuxeo.runtime.management.api.ProbeStatus;
+import org.nuxeo.runtime.services.config.ConfigurationService;
 
 import com.codahale.metrics.Counter;
 import com.codahale.metrics.MetricRegistry;
 import com.codahale.metrics.SharedMetricRegistries;
 
 /**
- * A probe to detect when computation has been terminated due to failure.
+ * A probe to detect when computation has been terminated due to failure. A delay is applied before returning the
+ * failure code.
  *
  * @since 11.1
  */
 public class StreamProbe implements Probe {
 
-    protected static final String FAILURE_MESSAGE = "%d computations have been terminated after failure. This Nuxeo instance must be restarted within the stream retention period.";
+    public static final String STREAM_PROBE_DELAY_PROPERTY = "nuxeo.stream.health.check.delay";
+
+    public static final Duration STREAM_PROBE_DELAY_DEFAULT = Duration.ofHours(36);
+
+    protected static final String FAILURE_MESSAGE = "%d computations have been terminated after failure. "
+            + "First failure detected: %s, probe failure delayed by %s. "
+            + "This Nuxeo instance must be restarted within the stream retention period.";
 
     protected Counter globalFailureCount;
 
+    protected Long detected;
+
+    protected Duration timeout;
+
     @Override
     public ProbeStatus run() {
         long failures = getFailures();
-        if (failures > 0) {
-            return ProbeStatus.newFailure(String.format(FAILURE_MESSAGE, failures));
+        if (failures == 0) {
+            return ProbeStatus.newSuccess("No failure");
+        }
+        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS");
+        String dateFailure = dateFormat.format(new Date(getDetected()));
+        String message = String.format(FAILURE_MESSAGE, failures, dateFailure, getTimeout().toString());
+        if ((System.currentTimeMillis() - getDetected()) < getTimeout().toMillis()) {
+            // Failure is delayed
+            return ProbeStatus.newSuccess(message);
+        }
+        return ProbeStatus.newFailure(message);
+    }
+
+    protected Duration getTimeout() {
+        if (timeout == null) {
+            ConfigurationService confService = Framework.getService(ConfigurationService.class);
+            timeout = confService.getDuration(STREAM_PROBE_DELAY_PROPERTY, STREAM_PROBE_DELAY_DEFAULT);
         }
-        return ProbeStatus.newSuccess("No failure");
+        return timeout;
+    }
+
+    protected long getDetected() {
+        if (detected == null) {
+            detected = System.currentTimeMillis();
+        }
+        return detected;
```

Comment:
Why not initializing `timeout` and `detected` in the no-args constructor?

Reply:
timeout requires the config service, doing it lazily removes the need to understand when the probe is created. Detected is the timestamp when the first failure is detected, I will refactor that so the value is not set in the getter.